core/goruntime.go

```diff
@@ -2,50 +2,63 @@ package core
 
 import (
 	"context"
-	"fmt"
-	"path/filepath"
+	"path"
 
 	"github.com/dagger/dagger/core/pipeline"
-	"github.com/moby/buildkit/client/llb"
 	bkgw "github.com/moby/buildkit/frontend/gateway/client"
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-func (p *Project) goRuntime(ctx context.Context, subpath string, gw bkgw.Client, platform specs.Platform) (*Directory, error) {
-	contextState, err := p.Directory.State()
+func (p *Project) goRuntime(ctx context.Context, gw bkgw.Client, progSock *Socket, pipeline pipeline.Path) (*Container, error) {
+	ctr, err := NewContainer("", pipeline, p.Platform)
+	if err != nil {
+		return nil, err
+	}
+	ctr, err = ctr.From(ctx, gw, "golang:1.20-alpine")
 	if err != nil {
 		return nil, err
 	}
 
 	workdir := "/src"
-	return NewDirectorySt(ctx,
-		goBase(gw).Run(llb.Shlex(
-			fmt.Sprintf(
-				`go build -o /entrypoint -ldflags '-s -d -w' %s`,
-				filepath.ToSlash(filepath.Join(workdir, filepath.Dir(p.ConfigPath), subpath)),
-			)),
-			llb.AddEnv("CGO_ENABLED", "0"),
-			llb.AddMount(workdir, contextState, llb.SourcePath(p.Directory.Dir)),
-			llb.Dir(workdir),
-			llb.AddMount(
-				"/go/pkg/mod",
-				llb.Scratch(),
-				llb.AsPersistentCacheDir("gomodcache", llb.CacheMountShared),
-			),
-			llb.AddMount(
-				"/root/.cache/go-build",
-				llb.Scratch(),
-				llb.AsPersistentCacheDir("gobuildcache", llb.CacheMountShared),
-			),
-		).Root(),
-		"",
-		pipeline.Path{},
-		platform,
-		nil,
-	)
-}
+	ctr, err = ctr.UpdateImageConfig(ctx, func(cfg specs.ImageConfig) specs.ImageConfig {
+		cfg.WorkingDir = absPath(cfg.WorkingDir, workdir)
+		cfg.Cmd = nil
+		return cfg
+	})
+	if err != nil {
+		return nil, err
+	}
+	ctr, err = ctr.WithMountedDirectory(ctx, gw, workdir, p.Directory, "")
+	if err != nil {
+		return nil, err
+	}
+
+	ctr, err = ctr.WithMountedCache(ctx, gw, "/go/pkg/mod", NewCache("gomodcache"), nil, CacheSharingModeShared, "")
+	if err != nil {
+		return nil, err
+	}
+	ctr, err = ctr.WithMountedCache(ctx, gw, "/root/.cache/go-build", NewCache("gobuildcache"), nil, CacheSharingModeShared, "")
+	if err != nil {
+		return nil, err
+	}
+
+	ctr, err = ctr.WithExec(ctx, gw, progSock, p.Platform, ContainerExecOpts{
+		Args: []string{
+			"go", "build", "-o", "/entrypoint", "-ldflags", "-s -d -w",
+			path.Join(workdir, path.Dir(p.ConfigPath)),
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	ctr, err = ctr.UpdateImageConfig(ctx, func(cfg specs.ImageConfig) specs.ImageConfig {
+		cfg.Entrypoint = []string{"/entrypoint"}
+		return cfg
+	})
+	if err != nil {
+		return nil, err
+	}
 
-func goBase(gw bkgw.Client) llb.State {
-	return llb.Image("golang:1.20-alpine", llb.WithMetaResolver(gw)).
-		Run(llb.Shlex(`apk add --no-cache file git openssh-client`)).Root()
```

Comment:
I see these deps are removed from Python and Go. Were they always redundant?

Reply:
There are now, but in the early days of cloak all the sdks were in a private github repo and we needed these to pull those.

These actually likely are still needed if a user has a dep on a private repo, but IIRC these packages were not sufficient in and of themselves and the other configs needed have been lost to the sands of time, so it probably makes sense to just leave this out for now and come back to it on a clean slate if/when needed in the future.